@here/harp-materials/lib/MapMeshMaterials.ts

```diff
@@ -715,6 +728,41 @@ export class FadingFeatureMixin implements FadingFeature {
     }
 }
 
+/**
+ * Class to apply the code necessary to customize the standard shader to remove the diffuse light
+ * component.
+ */
+export class ShadowFeatureMixin {
+    onBeforeCompile?: CompileCallback;
+    private m_removeDiffuseLight?: boolean;
+
+    protected applyShadowParameters(params?: ShadowFeatureParameters) {
+        this.m_removeDiffuseLight = params?.removeDiffuseLight;
+        if (this.m_removeDiffuseLight === true) {
+            this.onBeforeCompile = chainCallbacks(this.onBeforeCompile, shader => {
+                shader.fragmentShader = THREE.ShaderChunk.meshphysical_frag.replace(
+                    "#include <lights_physical_pars_fragment>",
+                    correctShadowChunk
+                );
+            });
+        }
+    }
+
+    /**
+     * The mixin classes should call this method to register the property [[removeDiffuseLight]]
+     */
+    protected addShadowProperty(): void {
+        Object.defineProperty(this, "removeDiffuseLight", {
+            get: () => {
+                return this.m_removeDiffuseLight;
+            },
+            set: val => {
```

Comment:
I don't know if we this patch intends to support changing of `removeDiffuseLight`.

Because in this state `material.removeDiffuseLight = ...` completly doesn't work ...
If no, remove `set` and document

If it's indeed supported, one must ensure that changing value  of `m_removeDiffuseLight` triggers recompilation of shader (now it's only done in `applyShadowParameters` which is called only in constructor).

Reply:
These aren't needed in particular, however at least the MapMeshStandardMaterial class needs to have:
```
    get removeDiffuseLight(): boolean {
        // to be overriden
        return false;
    }
    set removeDiffuseLight(val: boolean) {
        // to be overriden
    }
```
Otherwise, if I remove them, then three.js will give a warning asking why the given parameter is passed into the material.

Reply:
Please see latest commit, I have added a comment to these properties.